espnet2/train/preprocessor.py

```diff
@@ -935,119 +921,131 @@ def __call__(
+                    phn = pyopenjtalk.g2p(syllables[i]).split(' ')
+                    #if syllables[i][0] == "ヴ" and len(syllables[i]) == 2:
+                    #    phn.remove("u")
                 phn_num = len(phn)
-                cnt += phn_num
-
-                assert index < len(timeseq) 
-                assert index + phn_num -1 < len(timeseq)
-                start = int((timeseq[index, 0] - offset) * self.fs)
-                end = int((timeseq[index + phn_num - 1, 1] - offset) * self.fs) + 1
-                if end > nsamples:
-                    end = nsamples - 1
-                index += phn_num
-                pitchseq[start:end] = pitchs[i]
-            assert cnt == lab_len
-
-            pitchseq.astype(np.int64)
-            temposeq.astype(np.int64)
-            data.pop(self.midi_name)
-
-            data["score"] = pitchseq
-            data["tempo"] = temposeq
-
-        if self.singing_name in data:
-
-            if self.train and self.rirs is not None and self.noises is not None:
-                singing = data[self.singing_name]
-                nsamples = len(singing)
-
-                # singing: (Nmic, Time)
-                if singing.ndim == 1:
-                    singing = singing[None, :]
+                vowel_num = 0
+                for p in phn:
+                    if p in vowel:
+                        vowel_num += 1
+                if vowel_num == 2 and phn[0] != "k" and phn[0] != "g":
+                    phn_num -= 1
+                assert vowel_num <= 2
+                phn_cnt += phn_num
+                for _ in range(phn_num):
+                    midis.append(notemidis[i])
+                if phn_num == 1:
+                    xml_timeseq.append([notetimeseq[i][0], notetimeseq[i][1]])
+                elif phn_num == 2:
+                    t = notetimeseq[i][0] + (notetimeseq[i][1] - notetimeseq[i][0]) * 0.25
+                    xml_timeseq.append([notetimeseq[i][0], t])
+                    xml_timeseq.append([t, notetimeseq[i][1]])
+                elif phn_num == 3:
+                    t1 = notetimeseq[i][0] + (notetimeseq[i][1] - notetimeseq[i][0]) * 0.1
+                    t2 = notetimeseq[i][0] + (notetimeseq[i][1] - notetimeseq[i][0]) * 0.5
+                    xml_timeseq.append([notetimeseq[i][0], t1])
+                    xml_timeseq.append([t1, t2])
+                    xml_timeseq.append([t2, notetimeseq[i][1]])
+                elif phn_num == 4:
+                    t1 = notetimeseq[i][0] + (notetimeseq[i][1] - notetimeseq[i][0]) * 0.05
+                    t2 = notetimeseq[i][0] + (notetimeseq[i][1] - notetimeseq[i][0]) * 0.1
+                    t3 = notetimeseq[i][0] + (notetimeseq[i][1] - notetimeseq[i][0]) * 0.5
+                    xml_timeseq.append([notetimeseq[i][0], t1])
+                    xml_timeseq.append([t1, t2])
+                    xml_timeseq.append([t2, t3])
+                    xml_timeseq.append([t3, notetimeseq[i][1]])
```

Comment:
Can we add some hyper parameters to this split?

Reply:
Yes! The most ideal solution is to make the syllable-phonemes conversion using tokenizer. But for the mismatch problem we discussed before and the differences of phoneme formation between languages. It may not be easy to analyze into hyper parameters. If the mismatch problem and settings for different languages were settled, we could use hyper parameters here. For now, it only works on Japanese dataset. I will add a todo here.

Reply:
Sounds reasonable. Thanks for clarification